AirSkyBoat/AirSkyBoat
# Fix and rewrite of 'Like Shining' quests in Norg

**_I affirm:_**


## Please enter a player-facing description


- Fixed issue where the NPC Heizo would incorrectly report the number of Rusty Subligars turned in to Heiji during the "Like Shining Leggings" quest instead of Rusty Leggings. (HeavensSword)

## What does this pull request do? (Please be technical)


- The wrong player variable was being referenced in Heizo's Lua script ("shiningSubligar_nb" instead of "shiningLeggings_nb") when speaking with him while the quest "Like Shining Leggins" is active.
- EDIT: Rewrote Heizo's and Heiji's quests using the interaction framework as recommended. 

## Steps to test these changes


- Start the quest "Like Shining Leggings" by speaking with Heizo.
- Trade him X number of Rusty Leggings (less than 10)
- Speak with him and he should report 0. 
Note: If you start the quest "Like A Shining Subligar" with Heiji and trade him X number of Rusty Subligar, when speaking with Heizo again, he'll report that number.

## Special Deployment Considerations




Run these SQL updates to preserve existing Quest progression for players:

```
UPDATE char_vars SET varname = 'Quest[5][139]Prog' WHERE varname = "shiningLeggings_nb";
UPDATE char_vars SET varname = 'Quest[5][138]Prog' WHERE varname = "shiningSubligar_nb";
```
